packages/curve-ui-kit/src/lib/i18n.ts

```diff
@@ -55,14 +54,12 @@ export function parseLocale(locale?: string): { parsedLocale: Locale['value']; p
   }
 }
 
-export async function dynamicActivate(locale: string) {
-  const { parsedLocale } = parseLocale(locale)
-  let data = await import(`../locales/${parsedLocale}/messages`)
+export async function dynamicActivate(locale: string, data: any) {
   i18n.load(locale, data.messages)
   i18n.activate(locale)
 }
 
-export function updateAppLocale(locale: string, updateGlobalStoreByKey: <T>(key: DefaultStateKeys, value: T) => void) {
+export function updateAppLocale(locale: string, updateGlobalStoreByKey: <T>(key: 'locale', value: T) => void) {
```

Comment:
Unused parameter
```suggestion
export function updateAppLocale(locale: string, updateGlobalStoreByKey: <T>(value: T) => void) {
```

Reply:
I can't remove it as it's used below: `updateGlobalStoreByKey('locale', locale)`. It's an injected function whose type signature is `updateGlobalStoreByKey: <T>(key: DefaultStateKeys, value: T) => void`. I guess you can remove it but then you need to not forget and copypaste the `updateGlobalStoreByKey` invocation every time `updateAppLocale` is called.